dora/src/baseline/codegen.rs

```diff
@@ -1166,19 +1167,18 @@ where
             return;
         }
 
-        let (ty, field) = {
-            let ident_type = self.src.map_idents.get(expr.id).unwrap();
+        let ident_type = self.src.map_idents.get(expr.id).unwrap();
 
-            match ident_type {
-                &IdentType::Field(ty, field) => (ty, field),
-                _ => unreachable!(),
-            }
-        };
-
-        let ty = self.specialize_type(ty);
+        match ident_type {
+            &IdentType::Field(ty, field) => {
+                let ty = self.specialize_type(ty);
 
-        self.emit_expr(&expr.lhs, REG_RESULT.into());
-        self.emit_field_access(expr.pos, ty, field, REG_RESULT, dest.any_reg());
+                self.emit_expr(&expr.lhs, REG_RESULT.into());
+                self.emit_field_access(expr.pos, ty, field, REG_RESULT, dest.any_reg());
+            }
+            &IdentType::Method(ty, method) => unimplemented!(),
```

Comment:
This is where we currently fail – I'm unsure we should even end up with an `ExprDotType` and in `emit_dot` in the first place.